lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py

```diff
@@ -0,0 +1,346 @@
+            if old_profile['set_data']['notes'] != new_profile['notes']:
+                # existing profile has different notes
+                profile_dict['set_data'] = dict(notes=new_profile['notes'])
+
+        if new_profile['notes'] and ('set_data' not in old_profile or 'notes' not in old_profile['set_data']):
+            # existing profile has no notes, but we have notes
+            profile_dict['set_data'] = dict(notes=new_profile['notes'])
+
+        if profile_dict:
+            # if we have any updated values
+            changed = True
+            try:
+                result = self.client.post(old_profile['href'],
+                                          resource=profile_dict,
+                                          action="edit")
+            except Exception as e:
+                msg = "Updating profile '{description}' failed: {error}"
+                msg = msg.format(description=old_profile['description'], error=e)
+                self.module.fail_json(msg=msg, result=result)
+
+        if changed:
+            msg = "Profile {description} updated successfully".format(description=new_profile['description'])
+        else:
+            msg = "No update needed for profile {description}".format(description=new_profile['description'])
+        return dict(changed=changed, msg=msg)
+
+
+def main():
+    argument_spec = dict(
+        description=dict(type='str'),
+        resource_type=dict(type='str', choices=['Vm',
+                                                'ContainerNode',
+                                                'MiqServer',
+                                                'Host',
+                                                'Storage',
+                                                'EmsCluster',
+                                                'ExtManagementSystem',
+                                                'MiddlewareServer']),
+        alert_policies=dict(type='list'),
+        notes=dict(type='str'),
+        state=dict(default='present', choices=['present', 'absent', 'list']),
+    )
+    # add the manageiq connection arguments to the arguments
+    argument_spec.update(manageiq_argument_spec())
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_if=[('state', 'present', ['description', 'resource_type']),
+                                        ('state', 'absent', ['description'])])
+
+    state = module.params['state']
+    description = module.params['description']
+
+    manageiq = ManageIQ(module)
+    manageiq_alert_profiles = ManageIQAlertProfiles(manageiq)
+
+    if state == "list":
+        res_args = dict(changed=False, alerts=manageiq_alert_profiles.get_profiles())
+    else:
+        existing_profile = manageiq.find_collection_resource_by("alert_definition_profiles",
+                                                                description=description)
+
+        # we need to add or update the alert policy
```

Comment:
here again, and below L332, "alert policy" is not alert but typo for alert profile